plugins/obs-outputs/rtmp-stream.c

```diff
@@ -1360,9 +1399,19 @@ static void check_to_drop_frames(struct rtmp_stream *stream, bool pframes)
 			return;
 		}
 
-		if ((uint64_t)buffer_duration_usec >= DBR_TRIGGER_USEC) {
+		uint64_t t = os_gettime_ns();
+
+		if (buffer_duration_usec >= dbr_triggers[HIGH][1] && t >= stream->dbr_high_timeout) {
+			pthread_mutex_lock(&stream->dbr_mutex);
```

Comment:
Would try to simply all the duplicated code here.

Reply:
Moved the mutex lock and unlock to outside of the ifs. It did not made sense to have that lock and unlock code on each if there

Reply:
Sounds good